apps/api/src/app/bridge/e2e/sync.e2e.ts

```diff
@@ -561,93 +558,143 @@ describe('Bridge Sync - /bridge/sync (POST)', async () => {
   });
 
   it('should preserve control values across workflow syncs', async () => {
-    const workflowId = 'My Workflow';
-    const spaceSeparatedIdWorkflow = workflow(workflowId, async ({ step }) => {
-      await step.email('send-email', () => ({
-        subject: 'Welcome!',
-        body: 'Hello there',
-      }));
-    });
-    await bridgeServer.start({ workflows: [spaceSeparatedIdWorkflow] });
-
-    const firstSyncResponse = await session.testAgent.post(`/v1/bridge/sync`).send({
-      bridgeUrl: bridgeServer.serverPath,
-    });
-    expect(firstSyncResponse.body.data?.length).to.equal(1);
-
-    const firstWorkflowCountResponse = await workflowsRepository.count({ _environmentId: session.environment._id });
-    expect(firstWorkflowCountResponse).to.equal(1);
-
-    const firstWorkflowResponse = await workflowsRepository.findById(
-      firstSyncResponse.body.data[0]._id,
-      session.environment._id
+    const { workflowId, firstWorkflowResponse, firstSyncResponse } = await storeWorkflow(
+      bridgeServer,
+      session,
+      workflowsRepository
     );
 
-    expect(firstWorkflowResponse).to.be.ok;
-    if (!firstWorkflowResponse) {
-      throw new Error('Workflow not found');
-    }
+    await storeControlValue(session, workflowId, controlValuesRepository, firstWorkflowResponse);
 
-    expect(firstWorkflowResponse.name).to.equal(workflowId);
-    expect(firstWorkflowResponse.type).to.equal(WorkflowTypeEnum.BRIDGE);
-    expect(firstWorkflowResponse.rawData.workflowId).to.equal(workflowId);
-    expect(firstWorkflowResponse.triggers[0].identifier).to.equal(workflowId);
-
-    expect(firstWorkflowResponse.steps.length).to.equal(1);
-    expect(firstWorkflowResponse.steps[0].stepId).to.equal('send-email');
-    expect(firstWorkflowResponse.steps[0]._templateId).to.exist;
-
-    await session.testAgent.put(`/v1/bridge/controls/${workflowId}/send-email`).send({
-      variables: { subject: 'Hello World again' },
-    });
+    const secondWorkflowResponse = await updateWorkflow(
+      session,
+      bridgeServer,
+      workflowsRepository,
+      firstSyncResponse,
+      workflowId
+    );
 
-    const firstControlValueResponse = await controlValuesRepository.find({
-      _environmentId: session.environment._id,
-      _workflowId: firstWorkflowResponse._id,
-    });
+    await validateControlValues(controlValuesRepository, session, secondWorkflowResponse, workflowId);
```

Comment:
Please inline the expectations to the tests, it makes reading the test much easier without having to jump across functions.